dbt/adapters/snowflake/relation_configs/dynamic_table.py

```diff
@@ -69,12 +75,12 @@ def from_dict(cls, config_dict) -> "SnowflakeDynamicTableConfig":
             "query": config_dict.get("query"),
             "target_lag": config_dict.get("target_lag"),
             "snowflake_warehouse": config_dict.get("snowflake_warehouse"),
+            "catalog": SnowflakeCatalogConfig.from_dict(config_dict["catalog"]),
```

Comment:
Why not get? Is this guaranteed to exist?

Reply:
Yes, this is guaranteed to exist. Even if Iceberg is off, this defaults to the default table type on the catalog object. The describe query returns None for the entire catalog object, which is checked in the catalog parsing method. It's a bit clunky; the jinja template should really return 0 rows instead of a None object. But I'd rather put the logic in python than jinja.